File: FormView/Account/SignInController.m
//
//  SignInController.m
//  FormView
//
//  Created by 黄伯驹 on 2017/12/4.
//  Copyright © 2017年 黄伯驹. All rights reserved.
//

#import "SignInController.h"

#import "AccountFieldCell.h"
#import "AccountTitleCell.h"
#import "ButtonCell.h"
#import "PasswordLoginCell.h"
#import "AccountButtonCell.h"

@interface SignInController ()<ButtonCellActionable>

@end

@implementation SignInController

- (void)initSubviews {

    // 去掉顶部sectionHeader
    self.tableView.tableHeaderView = [[UIView alloc] initWithFrame:CGRectMake(0, 0, 0, 0.1)];

    AccountCellItem *accountItem = [[AccountCellItem alloc] init];
    accountItem.placeholder = @"请输入登录账号";

    AccountTitleCellItem *accountTitleCellItem = [[AccountTitleCellItem alloc] init];
    accountTitleCellItem.text = @"Sign In";

    ButtonCellItem *buttonCellItem = [[ButtonCellItem alloc] init];
    buttonCellItem.title = @"下一步";

    self.form = @[
                  @[
                      [[Row alloc] initWithClass:[AccountFieldCell class] model:accountItem],
                      [[Row alloc] initWithClass:[AccountFieldCell class] model:accountItem],
                      [[Row alloc] initWithClass:[AccountTitleCell class] model:accountTitleCellItem],
                      [[Row alloc] initWithClass:[PasswordLoginCell class]],
                      [[Row alloc] initWithClass:[ButtonCell class] model:buttonCellItem],
                      [[Row alloc] initWithClass:[AccountButtonCell class] model:buttonCellItem],
                      ]
                  ];
}

- (void)buttonAction:(UIButton *)sender {
    NSLog(@"class = %@   %@", self.classForCoder, sender.currentTitle);
}

@end

File: Swift/FormView/Controllers/SignInController.swift
//
//  SignInController.swift
//  FormView
//
//  Created by 黄伯驹 on 21/01/2018.
//  Copyright © 2018 黄伯驹. All rights reserved.
//

class SignInController: GroupTableController {
    override func initSubviews() {
        form = [
            [
                Row<TitleCell>(viewData: TitleCellItem(title: "Sign In")),
                Row<InputCell>(viewData: InputCellItem(placeholder: "请输入账号")),
                Row<InputCell>(viewData: InputCellItem(placeholder: "请输入密码")),
                Row<ButtonCell>(viewData: ButtonCellItem(title: "sign in", normalState: (.white, .normal), highlightState: (.lightGray, .highlighted))),
                Row<LicenseCell>(viewData: NoneItem())
            ]
        ]
    }
}

extension SignInController: ButtonCellActionable {
    func buttonAction(_ sender: UIButton) {
        
    }
}
